protocol/x/affiliates/keeper/keeper.go

```diff
@@ -41,3 +49,154 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 }
 
 func (k Keeper) InitializeForGenesis(ctx sdk.Context) {}
+
+func (k Keeper) RegisterAffiliate(
+	ctx sdk.Context,
+	referee string,
+	affiliateAddr string,
+) error {
```

Comment:
Should check both `refereee` and `affiliateAddr` are valid dYdX addresses with `sdk.AccAddressFromBech32. Return error accordingly